Documentation/git-diff.txt

```diff
@@ -81,7 +81,7 @@ in the last form, that use the "\..." notation, can be any
 <tree>. For instance, if you want to make diffs against an empty
 tree, you can create a tag pointing to the empty tree:
 
-'git tag' empty 4b825dc642cb6eb9a060e54bf8d69288fbee4904
+'git tag' empty $('git hash-object' -t tree /dev/null)
```

Comment:
I kind of see this as a pointless change, at least for two reasons. When we move away from the SHA-1 hash and adopt a different one, it is likely that we'd still understand an old-style object, like an empty tree with known object name, so it would be plausible that the given example would keep working after the transition. Also, the proposed change makes an assumption that the object payload for an empty tree will stay to be a zero-length string when we move away from SHA-1 hash, which is not warranted. The time we have to break backward compatibility is when we would make other changes, too.